### PR TITLE
otlp: mention service.name in kitchen sink cfg doc - closes #475

### LINKED ISSUE #475: opentelemetry-otlp how to name service: OTLPResourceNoAttributes
Currently, all my traces show up as top level `OTLPResourceNoAttributes`:

trace viewer screenshot

Is there a way to inject the service name with `opentelemetry-otlp` like in [opentelemetry-jaeger](https://docs.rs/opentelemetry-jaeger/0.11.0/opentelemetry_jaeger/struct.PipelineBuilder.html#method.with_service_name)?

### PR DESCRIPTION
**Pull request by clux**
adds a possibly slightly more useful key/value combination for docs.

it's probably a common thing to want to set, even if it's not a thing that otel cares about.

### CODE REVIEW

**Review by TommyCpp — APPROVED** [review]

**Review by TommyCpp — CHANGES_REQUESTED** [review]
May need to do the same change in README?

**Comment by clux** [comment]
Done.

Btw would you accept PR for adding `OtlpPipelineBuilder::with_service_name` like in [opentelemetry-jaeger](https://docs.rs/opentelemetry-jaeger/0.11.0/opentelemetry_jaeger/struct.PipelineBuilder.html#method.with_service_name)?

**Comment by TommyCpp** [comment]
> Btw would you accept PR for adding `OtlpPipelineBuilder::with_service_name` like in [opentelemetry-jaeger](https://docs.rs/opentelemetry-jaeger/0.11.0/opentelemetry_jaeger/struct.PipelineBuilder.html#method.with_service_name)?
> Might cater to the common use case a bit more ergonomically.

Hmm, I guess the problem here is OTLP still doesn't have the `service.name` concept, `with_service_name` would be confusing if users are using a backend that doesn't recognize the `service.name` resource. But maybe we should make a note in the doc to explain if the backend recognizes a Jaeger-style service name, we can set it via resource.

**Review by TommyCpp — APPROVED** [review]

File: opentelemetry-otlp/README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
                 .with_max_events_per_span(64)
                 .with_max_attributes_per_span(16)
                 .with_max_events_per_span(16)
-                .with_resource(Resource::new(vec![KeyValue::new("key", "value")])),
+                 with_resource(Resource::new(vec![KeyValue::new("service.name", "example")])),
```
Comment (jtescher) [inline opentelemetry-otlp/README.md:134]:
```suggestion
                 .with_resource(Resource::new(vec![KeyValue::new("service.name", "example")])),
```

**Review by jtescher — CHANGES_REQUESTED** [review]

File: opentelemetry-otlp/README.md (hunk 1 above)
Comment (clux) [reply opentelemetry-otlp/README.md]: ah, whoops :|

File: opentelemetry-otlp/README.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
                 .with_max_events_per_span(64)
                 .with_max_attributes_per_span(16)
                 .with_max_events_per_span(16)
-                .with_resource(Resource::new(vec![KeyValue::new("key", "value")])),
+                 .with_resource(Resource::new(vec![KeyValue::new("service.name", "example")])),
```
Comment (clux) [inline opentelemetry-otlp/README.md:134]: # #
```suggestion
                .with_resource(Resource::new(vec![KeyValue::new("service.name", "example")])),
```

**Review by jtescher — APPROVED** [review]

**Pull request merged**